### DISCUSSION TITLE
Rainbowkit in metamask browser - potential bug

### DISCUSSION THREAD

**Discussion by gskril** [rainbow-me/rainbowkit#627] [Q&A] [answered]
I'm having a super weird bug on my website that uses rainbowkit, and not sure if it's a rainbowkit problem or a metamask iOS app problem. Would appreciate any input.

ensfairy.xyz will not open in the metamask browser, showing an error about resolving an ENS name (screenshot below). I have no idea what this means since there is no wallet connected, but I noticed this is only happening on non-subdomains. So dev.ensfairy.xyz, which is currently the same exact build as the root domain, works totally fine.

The only reason I think this bug may be related to rainbowkit instead of metamask is because my older websites that don't use rainbowkit don't have this issue, even when just using the root domain. I noticed rainbowkit.com redirects to www so I wonder if this is something you guys are aware of?

![mm-rainbowkit-error](https://user-images.githubusercontent.com/35093316/182037217-145d3f3f-7feb-4cda-b089-ac557be251e9.jpeg)

**Comment by sreyemnayr** [comment, marked as answer]
This is because metamask is trying to resolve the .xyz domain extensions with ENS, which doesn't work on L2s. It's super annoying.

**Reply by gskril** [reply]
Ahh this makes sense. Good reason to use www subdomain in the future with DNS-in-ENS. Thanks!
